src/core/unit.c

```diff
@@ -6375,6 +6379,92 @@ Condition *unit_find_failed_condition(Unit *u) {
+                return sd_bus_error_setf(
+                                error,
+                                SD_BUS_ERROR_INVALID_ARGS,
+                                "Live mounting '%s' on '%s' for unit '%s' cannot be queued: live mounting not supported for unit type '%s'",
+                                src,
+                                dst,
+                                u->id,
+                                unit_type_to_string(u->type));
+        }
+
+        if (u->load_state != UNIT_LOADED) {
+                log_unit_debug(u, "Unit not loaded");
+                return sd_bus_error_setf(
+                                error,
+                                BUS_ERROR_NO_SUCH_UNIT,
+                                "Live mounting '%s' on '%s' for unit '%s' cannot be queued: unit not loaded",
+                                src,
+                                dst,
+                                u->id);
+        }
+
+        if (!UNIT_IS_ACTIVE_OR_RELOADING(unit_active_state(u))) {
+                log_unit_debug(u, "Unit not active");
+                return sd_bus_error_setf(
+                                error,
+                                BUS_ERROR_UNIT_INACTIVE,
+                                "Live mounting '%s' on '%s' for unit '%s' cannot be queued: unit not active",
+                                src,
+                                dst,
+                                u->id);
+        }
+
+        if (unit_active_state(u) == UNIT_REFRESHING) {
+                log_unit_debug(u, "Unit already live mounting");
+                return sd_bus_error_setf(
+                                error,
+                                BUS_ERROR_UNIT_BUSY,
+                                "Live mounting '%s' on '%s' for unit '%s' cannot be queued: another operation in progress",
+                                src,
+                                dst,
+                                u->id);
+        }
+
+        return UNIT_VTABLE(u)->can_live_mount(u, src, dst, error);
+}
+
+int unit_live_mount(
+                Unit *u,
+                const char *src,
+                const char *dst,
+                sd_bus_message *message,
+                MountInNamespaceFlags flags,
+                const MountOptions *options) {
+
+        assert(u);
+
+        /* Special return values:
+         *
+         *   -EOPNOTSUPP → live mounting not supported for this unit type
+         *   -EBUSY      → unit currently can't be changed since it's not running or not properly loaded, or
+         *                 has a job queued or similar
+         */
+
+        if (!UNIT_VTABLE(u)->live_mount)
+                return -EOPNOTSUPP;
+
+        if (u->load_state != UNIT_LOADED)
+                return -EHOSTDOWN;
+
+        if (u->job)
+                return -EBUSY;
+
+        if (unit_active_state(u) != UNIT_ACTIVE)
+                return -EBUSY;
```

Comment:
hmm this duplicates the tests from unit_can_live_mount() to some degree. Why not just call that here first?